src/main/java/org/apache/commons/lang3/reflect/TypeUtils.java

```diff
@@ -1531,12 +1543,15 @@ public static final ParameterizedType parameterizeWithOwner(final Type owner, fi
      * @param variables expected map keys
      * @return array of map values corresponding to specified keys
      */
+    @SuppressWarnings({"array.access.unsafe.high","compound.assignment.type.incompatible"})/*
+    #7 - result.length = variables.length, hence result[index++] is valid, also, index++ is a valid assignment for @IndexOrHigh("result") 
+    */
```

Comment:
This seems like a bug in the Index Checker. Can you write a minimal test case and report it in the issue tracker?